kernel/gic/src/gic/cpu_interface_gicv2.rs

```diff
@@ -25,45 +29,47 @@ mod offset {
 // enable group 1
 const CTLR_ENGRP1: u32 = 0b10;
 
-/// Enables routing of group 1 interrupts for the current CPU
-pub fn init(registers: &mut GicRegisters) {
-    let mut reg = registers.read_volatile(offset::CTLR);
-    reg |= CTLR_ENGRP1;
-    registers.write_volatile(offset::CTLR, reg);
-}
+impl CpuRegsP1 {
+    /// Enables routing of group 1 interrupts for the current CPU
+    pub fn init(&mut self) {
+        let mut reg = self.ctlr.read();
+        reg |= CTLR_ENGRP1;
+        self.ctlr.write(reg);
+    }
 
-/// Interrupts have a priority; if their priority
-/// is lower or equal to this one, they're queued
-/// until this CPU or another one is ready to handle
-/// them
-pub fn get_minimum_priority(registers: &GicRegisters) -> Priority {
-    u8::MAX - (registers.read_volatile(offset::PMR) as u8)
-}
+    /// Interrupts have a priority; if their priority
+    /// is lower or equal to this one, they're queued
+    /// until this CPU or another one is ready to handle
+    /// them
+    pub fn get_minimum_priority(&self) -> Priority {
+        u8::MAX - (self.prio_mask.read() as u8)
+    }
 
-/// Interrupts have a priority; if their priority
-/// is lower or equal to this one, they're queued
-/// until this CPU or another one is ready to handle
-/// them
-pub fn set_minimum_priority(registers: &mut GicRegisters, priority: Priority) {
-    registers.write_volatile(offset::PMR, (u8::MAX - priority) as u32);
-}
+    /// Interrupts have a priority; if their priority
+    /// is lower or equal to this one, they're queued
+    /// until this CPU or another one is ready to handle
+    /// them
```

Comment:
these docs need some improvement. I like the description of interrupt priorities, but you don't actually explain what the function does. Devs would expect something like:

```rust
/// Returns the minimum interrupt priority for the current CPU.
///
/// Interrupt priority is ...
```